---
title: "oscar_figures"
author: "Oscar Wilkins"
date: "14/03/2021"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(tidyverse)
library(DESeq2)
library(here)
library(ggrepel)
```

## Ribosome profiling quality control

Perform quality control to check that reads are periodic.

```{r message=FALSE, warning=FALSE}
read_length_filter <- 29  # length of reads of interest
gencode_v29_info <- read_tsv(paste0(here(), "/data/longest_proteincoding_transcript_hs_details.txt"))
ribo_df <- read_tsv(paste0(here(), "/data/riboseq/sh_tdpb.Aligned.toTranscriptome.out.bam.bed.gz"), 
                    col_names = c("transcript_id", "start0", "end", "strand")) %>%
    inner_join(gencode_v29_info, by = "transcript_id") %>%
    dplyr::filter(end-start0 == read_length_filter, strand == "+") %>%
    mutate(distance_from_start = start0-cds_start) %>%
    group_by(distance_from_start) %>%
    mutate(n_this_dist = n()) %>%
    dplyr::select(distance_from_start, n_this_dist) %>%
    unique()
    
```

```{r, warning=FALSE}
ggplot(ribo_df, aes(x=distance_from_start, y=n_this_dist)) +
    geom_bar(stat="identity") +
    xlim(-50, 50) +
    ylab("Counts") + 
    xlab("Distance of 5' end from annotated start codon") +
    ggtitle(paste0("Periodicity of ", read_length_filter, "nt reads from TDP-43 KD replicate B"))
```

## Ribosome profiling volcano plot

Use DESeq to detect differential ribosome footprints and create volcano plots

```{r message = FALSE, warning=FALSE}
counts_df <- read_csv(paste0(here(), "/data/riboseq/cds_feature_counts.csv"))
counts <- as.matrix(counts_df %>% column_to_rownames("gene_id"))
ipsc_splicing <- read_csv(paste0(here(), "/data/ipsc_splicing_results.csv"))
ipsc_de = read_csv(paste0(here(), "/data/ipsc_differential_expression.csv"))

col_data <- data.frame(sample = colnames(counts)) %>%
  mutate(condition = ifelse(str_detect(sample, "tdp"), "tdp_ko", "control"))

dds <- DESeqDataSetFromMatrix(countData = counts,
                              colData = col_data,
                              design= ~condition)
dds <- DESeq(dds)
results_name <- resultsNames(dds)[2]
res <- results(dds, name=results_name)

res_df <- data.frame(res) %>%
  rownames_to_column("gene_id") %>%
  left_join(gencode_v29_info, by = "gene_id")

cryptic_df <- ipsc_splicing %>%
  dplyr::select(gene_name, cryptic_junction) %>%
  unique() %>%
  filter(cryptic_junction)

res_df2 <- res_df %>%
  mutate(cryptic_junction = gene_name %in% cryptic_df$gene_name) %>%
  mutate(label_junction = case_when(gene_name %in% c("UNC13A","AGRN",
                                                     "UNC13B","PFKP","SETD5",
                                                     "ATG4B","STMN2","TARDBP") ~ gene_name)) %>%
  mutate(colour = ifelse(cryptic_junction, "#fe6101",  "#648FFF")) %>%
  arrange(colour)
```
   
```{r, warning=FALSE} 
p1 <- ggplot(res_df2, aes(x = log2FoldChange, y = -log10(pvalue), label = label_junction)) +
  geom_point(data = res_df2 %>% filter(!cryptic_junction),
             aes(x = log2FoldChange, y= -log10(pvalue)), color = "#648FFF") +
  geom_point(data = res_df2 %>% filter(cryptic_junction),
             aes(x = log2FoldChange, y= -log10(pvalue)), color = "#fe6101") +
  ggpubr::theme_pubr() +
  ylab("-Log10(p-value)") +
  xlab("Log2 fold change") +
  geom_text_repel(box.padding = 1.2, max.overlaps = Inf)

p2 <- p1 + ylim(0,10)
p1
p2
```

## Targeted RNA-seq

Read in data from processed bam files

```{r message=FALSE, warning=FALSE}
specific_rt <- read_csv(paste0(here(), "/data/targeted_RNA_seq/specific_RT_primer/for_plot.csv")) %>%
    select(risk, sample_name, n)

random_hexamer <- read_csv(paste0(here(), "/data/targeted_RNA_seq/random_hexamer/values_for_plot.csv")) %>%
    select(-sample_name) %>%
    select(sample_name = actual_sample_name, risk, n)

tg_rnaseq_df <- bind_rows(specific_rt, random_hexamer) %>%
  group_by(sample_name) %>%
  mutate(n_healthy = ifelse(risk == "Non-Risk", n, 0)) %>%
  mutate(combined_healthy = sum(n_healthy)) %>%
  select(-n_healthy) %>%
  mutate(combined = sum(n)) %>%
  group_by(sample_name, risk) %>%
  mutate(combined_n = sum(n)) %>%
  ungroup() %>%
  select(-n) %>%
  unique() %>%
  group_by(sample_name) %>%
  mutate(p_value = pbinom(combined_healthy, sum(combined_n), 0.5)) %>%  # single-tailed bionomial test
  mutate(colour = ifelse(risk == "Non-Risk", "lightblue", "red")) %>%
  ungroup() %>%
  mutate(sample_name_ordered = fct_reorder(sample_name, dplyr::desc(combined_n)))

ggplot(tg_rnaseq_df %>%
         filter(combined_n > 0), aes(x = sample_name_ordered, y = combined_n, fill = risk)) +
  geom_bar(stat="identity", position="dodge") +
  geom_text(aes(x=sample_name, y=-2.5, label=signif(p_value,2)), size = 3.5) +
  xlab("") +
  ylab("Unique cDNAs") +
  ggeasy::easy_add_legend_title("Allele") +
  theme_minimal() +
  ggpubr::theme_pubr() +
  ggeasy::easy_rotate_x_labels()
```

## iCLIP

##
