---
title: "Figures for Paper"
author: "Anna-Leigh Brown"
date: "06/10/2020"
output: 
    html_document:
        code_folding: hide
        toc: true
        toc_float: true
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE,warning=FALSE)
if (!require("pacman")) install.packages("pacman")
library(pacman)
p_load("data.table",
       "tidyverse",
       "ggplot2",
       "rcompanion",
       "lme4",
       "ggpubr",
       "forcats",
       "scales",
       "here",
       "ggthemes",
       "ggrepel")
```

```{r setup_function, echo=FALSE}
####helper funciton that gets the overall model p-value####
lmp <- function (modelobject) {
    if (class(modelobject) != "lm") stop("Not an object of class 'lm' ")
    f <- summary(modelobject)$fstatistic
    p <- pf(f[1],f[2],f[3],lower.tail=F)
    attributes(p) <- NULL
    return(p)
}

pretty_effect_plot = function(model, p_value_cutoff = 0.05, title = "Model 1"){
  # Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
    p_value = lmp(model)
    r_squared = round(summary(model)$adj.r.squared,3)
    pretty_plot = broom::tidy(model) %>% 
        mutate(p_star = case_when(p.value < 0.001 ~ "***",
                                  p.value < 0.01 ~ "**",
                                  p.value < 0.05 ~ "*",
                                  p.value < 0.1 ~ ".",
                                  p.value > 0.1 ~ "NS")) %>% 
        filter(p.value < p_value_cutoff) %>%
        mutate(term = fct_reorder(term, -abs(estimate))) %>%
        ggplot(aes(x = term, y = estimate,fill = estimate > 0,show.legend = F)) +
        geom_col() +
        coord_flip() +
        ggtitle(title,subtitle = glue::glue("Model p-value: {p_value} \n Adj. R-squared: {r_squared}")) +
        geom_text(aes(x=term, y= -0.01, label=p_star)) 
}
```
# iPSC splicing

Read in the splicing results returned by MAJIQ and make a volcano plot, only highlight 
genes of interest with a label.

```{r}
junc_colors = c("#648FFF","#fe6101")
ipsc_splicing = fread(file.path(here(),"data","ipsc_splicing_results.csv")


ipsc_splicing %>% 
  mutate(junction_name = glue::glue("{gene_name} - {paste_into_igv_junction}")) %>% 
  mutate(`Novel Junction` = de_novo_junctions == 0) %>% 
  mutate(log10_test_stat = -log10(1 - p_d_psi_0_10_per_lsv_junction)) %>% 
  mutate(log10_test_stat = ifelse(is.infinite(log10_test_stat), 16, log10_test_stat)) %>% 
  mutate(graph_alpha = ifelse(p_d_psi_0_10_per_lsv_junction > 0.9, 1, 0.2)) %>% 
  mutate(label_junction = case_when(gene_name %in% c("UNC13A",
                                                     "UNC13B","PFKP","SETD5",
                                                     "ATG4B","STMN2") & 
                                      p_d_psi_0_10_per_lsv_junction > 0.9  & 
                                      deltaPSI > 0 ~ junction_name,
                                    T ~ "")) %>% 
  ggplot(aes(x = deltaPSI, y = log10_test_stat)) +
  geom_point(aes(fill = `Novel Junction`,alpha = graph_alpha), pch = 21, size = 4) + 
  geom_text_repel(aes(label = label_junction), point.padding = 0.3,
                  min.segment.length = unit(0, 'lines')) +
  geom_hline(yintercept = -log10(1 - .9)) + 
  guides(alpha = FALSE, size = FALSE) + 
  scale_fill_manual(values = junc_colors) + 
  theme(legend.position = 'top') + 
  ggpubr::theme_pubr() + 
  xlab("delta PSI") + 
  ylab("-Log 10 Test Statistic") + 
  theme(text = element_text(size = 18))
```

# UNC13A Cryptic PSI across TDP-43 KD experiments

We have 2 PSI's for the UNC13A cryptic, one which includes both the short and
long form of the cryptic, and one which does not. While we're not mentioning in 
figure 1, given that the longer form of the cryptic appears in control cerebellum, 
one could argue that that junction should not be included in the PSI calculation
here for UNC13A cryptic per se. In practice, this makes very little difference in the
conclusions made on the cell lines, so I've included both for completeness.

The 'C/G' tells which genotypes were supported by RNA-seq on rs12973192. 
The NB cell lines are het, as is the WTC11 cell line. SH-SY5Y cells are homozygote for the major allele. 
There was variability on the Klim hMN set on allelic expression. 
```{r}


rel_rna_cryptic_amount = fread(file.path(here(),"data","kd_experiments_relative_rna_and_unc13a_cryptic_junction_counts.csv"))
rel_rna_cryptic_amount[,cryptic_psi_full := ( UNC13A_3prime + 
                                          UNC13A_5prime +  UNC13A_5prime_2 + 
                                          UNC13A_5prime_3) / (UNC13A_annotated +  UNC13A_3prime + 
                                          UNC13A_5prime +  UNC13A_5prime_2 + 
                                          UNC13A_5prime_3)]
rel_rna_cryptic_amount %>% 
    ggbarplot(,
              x = "source",
              add = c("mean_se","jitter"),
              y = "cryptic_psi_full",
              fill = 'condition',
              color = 'condition',
              position = position_dodge(0.8)) + 
    ggpubr::theme_pubr() +
    scale_fill_manual(
        values = c("#40B0A6","#E1BE6A")
    ) + 
    scale_color_manual(
        values = c("#1C2617","#262114")
    ) +
    scale_x_discrete(labels=c("shsy5y_normed" = "NB cells \n C/G", 
                              "ipsc_normed_ward" = "iPSC Neurons \n C/G",
                              "shsy5y_normed_no_snp" = "SH-SY5Y \n C/C",
                              "klim_normed" = "Klim iPSC MN \n C/G + C/C",
                              "sh_dzap" = "Appocher \n NB cells \n C/G")) +
    ylab("UNC13A Cryptic PSI") + 
    xlab("") + 
    guides(color = FALSE) 

 rel_rna_cryptic_amount %>% 
    ggbarplot(,
              x = "source",
              add = c("mean_se","jitter"),
              y = "cryptic_psi",
              fill = 'condition',
              color = 'condition',
              position = position_dodge(0.8)) + 
    ggpubr::theme_pubr() +
    scale_fill_manual(
        values = c("#40B0A6","#E1BE6A")
    ) + 
    scale_color_manual(
        values = c("#1C2617","#262114")
    ) +
    scale_x_discrete(labels=c("shsy5y_normed" = "NB cells \n C/G", 
                              "ipsc_normed_ward" = "iPSC Neurons \n C/G",
                              "shsy5y_normed_no_snp" = "SH-SY5Y \n C/C",
                              "klim_normed" = "Klim iPSC MN \n C/G + C/C",
                              "sh_dzap" = "Appocher \n NB cells \n C/G")) +
    ylab("UNC13A Cryptic PSI") + 
    xlab("") + 
    guides(color = FALSE) 
```

# Paient summary statistics
First let's look at only including tissues with detectable stmn2 or UNC13A CE
```{r}
clean_data_table = fread(file.path(here(),"data","nygc_junction_information.csv"))
clean_data_table = clean_data_table %>%     
    mutate(call = fct_relevel(call,
                              "C/C", "C/G", "G/G")) %>% 
    mutate(number_g_alleles = as.numeric(call) - 1) %>% 
    mutate(unc13a_cryptic_leaf_psi = ifelse(is.na(unc13a_cryptic_leaf_psi),0,unc13a_cryptic_leaf_psi)) %>% 
    mutate(stmn2_psi_groups = as.numeric(cut_interval(log10(stmn_2_cryptic_psi_leaf), n = 2))) %>%
    mutate(stmn2_psi_groups = case_when(stmn2_psi_groups == 1 ~ " Low STMN2",
                                      stmn2_psi_groups == 2 ~ "High STMN2",
                                      TRUE ~ "No STMN2"))

print(glue::glue("Number of unique patients: {clean_data_table[,length(unique(participant_id))]}"))
print(glue::glue("Number of unique tissue samples: {clean_data_table[,length(unique(sample))]}"))
print("Patients Per Disease Category")
clean_data_table[,length(unique(participant_id)),by = disease]
print("Tissues Per Disease Category")
clean_data_table[,length(unique(sample)),by = disease]
print("Number of patients per rs12973192 genotype")
clean_data_table[,length(unique(participant_id)),by = call]
print("Number of tissues per disease")
clean_data_table[,.N,by = c("disease","tissue_clean")]
print("Number of partcipants by mutation and  disease")
clean_data_table[,length(unique(participant_id)),by = c("mutations","disease")]

print(glue::glue("Number of patients per pathology:"))
clean_data_table[,length(unique(participant_id)),by = .(pathology)]
```
# UNC13A cryptic is an event that is specific to TDP-43 proteinopathy

FTLD-non-TDP are those with TAU and FUS aggregates

Non-tdp ALS are those with SOD1 or FUS mutations. The n's are quite low on this 
unfortunately, only 8 ALS with SOD1 and 2 with FUS mutations.  

First we look at detection rate in tissues affected by TDP-43 proteinopathy,
For FTLD this is frontal and temporal Cortices, and
for ALS this is lumbar, cervical, and thoracic spinal cord samples as well as 
motor cortex. For controls we also take all 6 tissues, 
frontal,temporal,motor cortices and the lumbar, cervical, and thoracic spinal cords. 

(As a side note, once we do this the number of ALS-non-TDP drops down to 6 (2 FUS) because
the ALS sample tissues are not balanced and not every participant has samples in every tissue)
```{r}
####Inclusion reads by if TDP-potential####
boxplot_table = clean_data_table %>% 
    mutate(disease_group2 = case_when(disease == "Control" ~ "Control",
                                 pathology %in% c("FTD-TDP-A","FTD-TDP-B","FTD-TDP-C") ~ "FTLD-TDP",
                                 pathology %in% c("FTD-TAU","FTD-FUS") ~ "FTLD \n non-TDP",
                                 mutations %in% c("SOD1","FUS") ~ "ALS \n non-TDP",
                                 T ~ "ALS-TDP")) %>% 
    mutate(across(UNC13A_3prime_leaf:UNC13A_annotated_leaf, ~ .x / library_size,.names = "{.col}_library_norm")) %>% 
    filter(!tissue_clean %in% c("Choroid","Liver")) %>% 
    dplyr::select(sample,participant_id,mutations,disease_group2,pathology,tissue_clean,contains("_library_norm")) %>% 
    melt() %>% 
    filter(grepl("_3prime|_5prime_",variable)) %>% 
    group_by(sample) %>% 
    mutate(inclusion_reads = sum(value)) %>% 
    ungroup() %>% 
    unique() %>% 
    mutate(junction_name = case_when(variable == "UNC13A_3prime_leaf_library_norm" ~ "  Novel 3'", 
                                 variable == "UNC13A_5prime_1_leaf_library_norm" ~ " Short Novel 5'", 
                                 variable == "UNC13A_5prime_2_leaf_library_norm"~ "Long Novel 5'")) %>% 
    mutate(disease_tissue = case_when((grepl("FTLD",disease_group2) & grepl("Cortex",tissue_clean))  ~ T,
                                      (grepl("ALS",disease_group2) & grepl("Cord|Motor",tissue_clean))  ~ T,
                                      (grepl("Occipital",tissue_clean)) ~ F,
                                      (grepl("Control",disease_group2) & grepl("Cord|Cortex",tissue_clean)) ~ T,
           TRUE ~ F)) %>% 
    mutate(tissue_clean = gsub("_"," ",tissue_clean))


```

Looking at disease tissue only, so just taking the cords in ALS and the 
frontal and temporal cortex of FTLD and then the cord and cortices in Controls.
```{r}

boxplot_table %>% 
  filter(disease_tissue == T) %>% 
  mutate(detected = inclusion_reads > 0) %>% 
  dplyr::select(participant_id,disease_group2,detected) %>% 
  unique() %>% 
  group_by(disease_group2) %>% 
  mutate(n_sample = n_distinct(participant_id)) %>% 
  mutate(n_sample_detected = sum(detected)) %>% 
  dplyr::select(disease_group2,n_sample,n_sample_detected) %>% 
  unique() %>% 
  mutate(detection_rate = n_sample_detected / n_sample) %>% 
  mutate(disease_group2 = gsub("Control"," Control",disease_group2)) %>% 
  mutate(detection_name = glue::glue("{disease_group2} \n ( {n_sample} )")) %>% 
  ggplot() +
  geom_col(aes(x = detection_name, y = detection_rate)) + 
  ggpubr::theme_pubr() + 
  scale_y_continuous(lim = c(0,1),labels = scales::percent) + 
  ylab("Percent of Patients \n UNC13A Cryptic Detected") + 
  theme(text = element_text(size = 18)) + 
  xlab("N individuals")


disease_comparisons = list( c("Control","ALS-TDP"),
                           c("Control","ALS \n non-TDP"),
                           c("Control","FTLD-TDP"),
                           c("Control","FTLD \n non-TDP" ))

table_to_test = boxplot_table %>% 
    filter(disease_tissue == T) %>% 
    group_by(disease_group2) %>% 
    mutate(n_sample = n_distinct(sample)) %>% 
    mutate(disease_group2 = gsub("Control"," Control",disease_group2)) %>% 
    mutate(detection_name = glue::glue("{disease_group2} \n ( {n_sample} )")) %>% 
    dplyr::select(detection_name,
                  participant_id,
                  inclusion_reads,
                  disease_group2) %>% 
    unique() 

test_pair = pairwise.wilcox.test(table_to_test$inclusion_reads, table_to_test$detection_name,
                     p.adjust.method = "BH") %>% broom::tidy()

test_pair = test_pair %>% 
  mutate(p_value_draw = case_when(p.value < 0.0001~ "***",
                                  p.value < 0.01 ~ "**",
                                   p.value < 0.05 ~ "*",
                          TRUE ~ paste0("Adj. p-value \n",as.character(round(p.value,2))))) %>% 
  mutate(y.position = seq(0.25,by = 0.1,length.out = 9))

table_to_test %>% 
    ggplot(aes(x = detection_name, y = inclusion_reads * 10^6)) + 
    geom_boxplot() + 
    geom_jitter(height = 0) + 
    scale_y_log10() + 
    ggpubr::theme_pubr() +
    ylab("UNC13A cryptic inclusion \n reads per million") + 
    theme(text = element_text(size = 18)) + 
    xlab("N samples") + 
    stat_pvalue_manual(test_pair %>% filter(p.value < 0.05),
                       label = "p_value_draw") +
   stat_compare_means()


####Inclusion reads by if TDP-potential####
boxplot_table %>% 
  filter(disease_group2 %in% c("Control",
                              "FTLD \n non-TDP",
                              "FTLD-TDP")) %>% 
    filter(tissue_clean %in% c("Cerebellum","Frontal Cortex","Temporal Cortex")) %>% 
  dplyr::select(disease_group2,
                tissue_clean,
                junction_name,
                value) %>% 
  unique() %>% 
    ggplot(aes(x = disease_group2, 
               y = value * 10^6,
               fill = junction_name)) + 
    geom_boxplot(show.legend = F) + 
    geom_jitter(height = 0,show.legend = F) + 
    scale_y_log10() + 
    ggpubr::theme_pubr() +
    facet_grid(vars(junction_name),vars(tissue_clean)) + 
    ylab("UNC13A cryptic \n reads per million") +
    theme(text = element_text(size = 18)) + 
    xlab("") +
    scale_fill_manual(values = colorblind_pal()(4)[2:4]) 
      


```

Splitting by the type of junction in ALS
```{r}
####Inclusion reads by if TDP-potential####
boxplot_table %>% 
  filter(disease_group2 %in% c("Control",
                              "ALS \n non-TDP",
                              "ALS-TDP")) %>% 
      filter(tissue_clean %in% c("Cerebellum","Motor Cortex","Temporal Cortex") | grepl("Cord",tissue_clean)) %>% 
  dplyr::select(disease_group2,
                tissue_clean,
                junction_name,
                value) %>% 
  unique() %>% 
  mutate(disease_group2 = fct_relevel(disease_group2,"Control","ALS \n non-TDP","ALS-TDP")) %>% 
    ggplot(aes(x = disease_group2, 
               y = value * 10^6,
               fill = junction_name)) + 
    geom_boxplot(show.legend = F) + 
    geom_jitter(height = 0,show.legend = F) + 
    scale_y_log10() + 
    ggpubr::theme_pubr() +
    facet_grid(vars(junction_name),vars(tissue_clean)) + 
    ylab("UNC13A cryptic \n reads per million") +
    theme(text = element_text(size = 18)) + 
    xlab("") +
    scale_fill_manual(values = colorblind_pal()(4)[2:4]) +
  theme(axis.text.x =  element_text(size = 14))
```

```{r}
clean_data_table %>% 
    mutate(disease_group2 = case_when(disease == "Control" ~ "Control",
                                 pathology %in% c("FTD-TDP-A","FTD-TDP-B","FTD-TDP-C") ~ "FTLD-TDP",
                                 pathology %in% c("FTLD-TAU","FTLD-FUS") ~ "FTLD-non-TDP",
                                 mutations %in% c("SOD1","FUS") ~ "ALS \n non-TDP",
                                 T ~ "ALS-TDP")) %>% 
    filter(disease_group2 %in% c("ALS \n non-TDP","ALS-TDP")) %>% 
    mutate(across(UNC13A_3prime_leaf:UNC13A_annotated_leaf, ~ .x / library_size,.names = "{.col}_library_norm")) %>% 
    filter(!tissue_clean %in% c("Choroid","Liver")) %>% 
    dplyr::select(sample,disease_group2,tissue_clean,contains("_library_norm")) %>% 
    melt() %>% 
    filter(grepl("_3prime|_5prime_1",variable)) %>% 
    group_by(sample) %>% 
    mutate(inclusion_reads = sum(value)) %>% 
    ungroup() %>% 
    dplyr::select(-variable,-value) %>% 
    unique() %>% 
    ggplot(aes(x = disease_group2, y = inclusion_reads * 10^6)) + 
    geom_boxplot() + 
    geom_jitter(height = 0) + 
    facet_wrap(~tissue_clean,nrow = 2) + 
    scale_y_continuous(trans = 'log10') + 
    ggpubr::theme_pubr() +
    ylab("UNC13A cryptic inclusion reads per million")  
```

# Relationship between UNC13A and STMN2 cryptic PSI

We noted in the cell lines that there was a linear relationship between the amount
of TARDBP KD efficiency and the amount of observed UNC13A cryptic. As a quantitative measure
of TDP-43 proteinopathy does not exist for these patients, no ranked staining, we used the amount of STMN2 
present in a tissue to rank the tissue by the amount of TDP-43 proteinopathy, as we've already shown
you can use that as a measure. Here we show the relationship in samples with 
at least 10 STMN2 annotated junctions and tissues where the PSI of UNC13A cryptic was also detectable, as measured by
STMN2 cryptic occurring in the tissue. 

## Scatter plot showing the relationship in non-logspace between STMN2 and UNC13A cryptic PSI
```{r}
####scatter plot showing the correlation in non-log space in STMN2 and cryptic PSI####
clean_data_table %>% 
    filter(STMN2_annotated_leaf > 10) %>% 
    filter(stmn_2_cryptic_psi_leaf > 0 ) %>% 
    filter(unc13a_cryptic_leaf_psi > 0 ) %>%
    group_by(call) %>% 
    ggpubr::ggscatter(., 
                      x = "stmn_2_cryptic_psi_leaf",
                      y = "unc13a_cryptic_leaf_psi",
                      add = "reg.line",
                      cor.coef = T) +
    ylab("UNC13A Cryptic PSI ") + 
    xlab("STMN2 Cryptic PSI ") 

####scatter plot showing the correlation in non-log space in STMN2 and cryptic PSI####
clean_data_table %>% 
    filter(STMN2_annotated_leaf > 10) %>% 
    filter(stmn_2_cryptic_psi_leaf > 0 ) %>% 
    filter(unc13a_cryptic_leaf_psi > 0 ) %>%
    mutate(call = fct_relevel(call,"C/C", "C/G", "G/G")) %>%
    group_by(call) %>% 
    ggpubr::ggscatter(., 
                      x = "stmn_2_cryptic_psi_leaf",
                      y = "unc13a_cryptic_leaf_psi",
                      add = "reg.line",
                      color = "call") +
    ylab("UNC13A Cryptic PSI ") + 
    xlab("STMN2 Cryptic PSI ") +
    stat_cor(aes(color = call),show.legend = FALSE) +
    scale_color_manual(values = c("#88CCEE","#44AA99","#105e2a"))
```

You can see a clear relationship, which appears to be modulated by the genotype
of rs12973192. This lead us to see if genotype was actually predictive of the 
amount of UNC13A cryptic psi. Because PSI's tend to be log linear, we chose
to model UNC13A in the log10 + 1 (log10(1) is zero) and we selected for tissues
which show TDP-43 pathology, as we assume this is an event which only begins to
show clear after TDP-43 pathology arises. 

## Scatter plot showing the relationship in log + 1 between STMN2 and UNC13A cryptic PSI

```{r}

clean_data_table %>% 
  filter(disease_tissue == T) %>% 
  filter(disease_group2 %in% c("FTLD-TDP","ALS-TDP")) %>% 
    mutate(call = fct_relevel(call,
                              "C/C", "C/G", "G/G")) %>%
    filter(STMN2_annotated_leaf > 10) %>% 
    filter(stmn_2_cryptic_psi_leaf > 0 ) %>% 
    filter(unc13a_cryptic_leaf_psi > 0 ) %>% 
    mutate(log10_stmn_2_cryptic_psi_leaf_plusone = log10(stmn_2_cryptic_psi_leaf + 1)) %>% 
    mutate(log10_unc13a_cryptic_leaf_psi_plusone = log10(unc13a_cryptic_leaf_psi + 1)) %>% 
    group_by(call) %>% 
    ggpubr::ggscatter(., 
                      x = "log10_stmn_2_cryptic_psi_leaf_plusone",
                      y = "log10_unc13a_cryptic_leaf_psi_plusone",
                      color = 'call',fill = 'call',                      
                      add = "reg.line") +
    ylab("Log10 UNC13A Cryptic PSI + 1") + 
    xlab("Log10 STMN2 Cryptic PSI + 1 ") +
    scale_color_manual(values = c("#88CCEE","#44AA99","#105e2a")) + 
    scale_fill_manual(values = c("#88CCEE","#44AA99","#105e2a"))


```

We can perform this analysis separated by tissue, but low numbers
make the results more complicated to interpret. Additionally, the tissue only
matters in so much as we know that certain tissues in certain diseases
are affected by TDP-43 pathology, whereas others aren't, tissue is not the 
most interesting or important variable, TDP-43 pathology is the question. 


```{r}
####scatter plot showing the correlation in non-log space in STMN2 and cryptic PSI####
clean_data_table %>% 
    filter(tissue_clean%in% c("Cervical_Spinal_Cord",
                              "Motor_Cortex",
                              "Frontal_Cortex","Temporal_Cortex")) %>% 
    mutate(call = fct_relevel(call,
                              "C/C", "C/G", "G/G")) %>%
    filter(STMN2_annotated_leaf > 10) %>% 
    filter(cryptic_psi > 0 ) %>% 
    filter(stmn_2_cryptic_psi > 0 ) %>% 
    group_by(call) %>% 
    ggpubr::ggscatter(., 
                      x = "stmn_2_cryptic_psi",
                      y = "unc13a_cryptic_leaf_psi",
                      color = 'call',                      
                      add = "reg.line") +
    ylab("UNC13A Cryptic PSI ") + 
    xlab("STMN2 Cryptic PSI ") +
    scale_color_manual(values = c("#88CCEE","#44AA99","#105e2a")) +
    facet_wrap(~tissue_clean)

```

## Modeling UNC13A cryptic using glms and mixed models

This brought us to the idea that we could even potentially model
UNC13A cryptic pathology using genotype and the amount of TDP-43 pathology. 
We've coded the number of G alleles here as a number and are looking into tissue
with TDP-43 pathology(measure with STMN2 cryptic PSI). 

```{r}

new_regress = clean_data_table %>%
    mutate(call = fct_relevel(call,
                              "C/C", "C/G", "G/G")) %>%
    filter(STMN2_annotated_leaf > 10) %>% 
    filter(stmn_2_cryptic_psi_leaf > 0 ) %>%  
    mutate(log10_stmn_2_cryptic_psi_leaf = log10(stmn_2_cryptic_psi_leaf)) %>% 
    mutate(log10_unc13a_cryptic_leaf_psi_plusone = log10(unc13a_cryptic_leaf_psi + 1)) %>% 
    mutate(log10_stmn_2_cryptic_psi_leaf_plusone = log10(stmn_2_cryptic_psi_leaf + 1))


another_model2 = lm(log10(unc13a_cryptic_leaf_psi + 1) ~ 
                   log10(stmn_2_cryptic_psi_leaf + 1) * number_g_alleles + pathology + tissue_clean,
              new_regress)


the_pretty_plot = pretty_effect_plot(another_model2,p_value_cutoff = 1,"UNC13A Cryptic PSI") + ggpubr::theme_pubr()
print(the_pretty_plot)
```

We find that at a cutoff of 0.01 for significant effects, only the interaction
between the number of G alleles and the amount of STMN2 pathology is predictive
of UNC13A cryptic inclusion. the absolute amount of STMN2 pathology does not
vary between genotypes. 
```{r}
genotype_comparisons = list(c("C/C", "C/G"), c("C/C", "G/G"), c("C/G", "G/G"))


ggboxplot(new_regress, x = "call", y = "log10_stmn_2_cryptic_psi_leaf",
          color = "call", palette = "jco")+
  stat_compare_means() +
  stat_compare_means(comparisons = genotype_comparisons)

ggboxplot(new_regress, x = "call", y = "stmn_2_cryptic_psi_leaf",
          color = "call", palette = "jco")+
  stat_compare_means() +
  stat_compare_means(comparisons = genotype_comparisons)

new_regress %>% 
    ggplot() + 
    stat_ecdf(aes(color = call, x = log10_stmn_2_cryptic_psi_leaf))
```

We see that detection rate by genotype varies. 

```{r}
overall_fisher = clean_data_table %>%
    filter(disease_tissue == T) %>% 
  filter(disease_group2 %in% c("FTLD-TDP","ALS-TDP")) %>% 
  mutate(unc13a_detected = unc13a_cryptic_leaf_psi > 0) %>% 
  dplyr::select(participant_id,unc13a_detected,call) %>% 
  unique() %>% 
  group_by(call) %>% 
  mutate(n_sample = n_distinct(participant_id)) %>% 
  mutate(n_sample_detected = sum(unc13a_detected)) %>% 
  dplyr::select(call,n_sample,n_sample_detected) %>% 
  unique() %>% 
  mutate(n_non_detected = n_sample - n_sample_detected) %>% 
  dplyr::select(-n_sample)


over_p = overall_fisher %>% 
  column_to_rownames('call') %>% 
  fisher.test() %>% 
  broom::tidy() %>% 
  .$p.value

overall_fisher %>% 
  mutate(n_sample = n_non_detected + n_sample_detected) %>% 
  mutate(detection_rate = n_sample_detected / n_sample) %>% 
  mutate(detection_name = glue::glue("{call} \n ( {n_sample} )")) %>% 
  ggplot(aes(x = detection_name, y = detection_rate, fill = detection_name)) +
  geom_col(show.legend = F) + 
  ggpubr::theme_pubr() + 
  scale_y_continuous(labels = scales::percent) + 
  ylab("% of TDP-43 Proteionopathy Patients \n UNC13A Cryptic Detected") + 
  theme(text = element_text(size = 18)) + 
  xlab("N individuals") + 
  scale_fill_manual(values = c("#88CCEE","#44AA99","#1B7739")) 

```


Although this difference is not significant, with the Fisher's exact giving
a p-value of `r round(over_p,2)`.


Only looking at disease relevant tissue and in patients 
we see that the detection rate depends on both the level of TPD-43 pathology present, 
again I acknowledge STMN2 is a proxy, but it is what is is with the dataset, and the geneotype 
of the patients. 

This difference is also not significant with a fisher's exact test, but
the 

```{r}
detection_table = clean_data_table %>% 
  filter(disease_tissue == T) %>% 
  filter(disease_group2 %in% c("FTLD-TDP","ALS-TDP")) %>% 
    mutate(call = fct_relevel(call,
                              "C/C", "C/G", "G/G")) %>%
    mutate(stmn2_psi_groups = as.numeric(cut_interval(log10(stmn_2_cryptic_psi), n = 2))) %>%
    mutate(stmn2_psi_groups = ifelse(is.na(stmn2_psi_groups),"  No STMN2",stmn2_psi_groups)) %>%
    mutate(stmn2_psi_groups = case_when(stmn2_psi_groups == 1 ~ " Low STMN2",
                                        stmn2_psi_groups == 2 ~ "High STMN2",
                                        TRUE ~ stmn2_psi_groups)) %>%
  mutate(unc13a_detected = unc13a_cryptic_leaf_psi > 0) %>% 
  group_by(call,unc13a_detected,stmn2_psi_groups) %>% 
  add_count(name = "genotype_detected") %>% 
  dplyr::select(call,unc13a_detected,genotype_detected,stmn2_psi_groups) %>% 
  unique() %>% 
  pivot_wider(names_from = "unc13a_detected",
              values_from = "genotype_detected") %>% 
  dplyr::rename(unc_not_detected = `FALSE`, unc_cryptic_detected = `TRUE`) %>%
  mutate(total_tissue = (unc_not_detected + unc_cryptic_detected)) %>% 
  mutate(detection_rate = unc_cryptic_detected / total_tissue) %>% 
  ungroup() %>% 
  mutate(stmn2_psi_groups = fct_relevel(stmn2_psi_groups, "No STMN2", after = 0)) %>% 
  mutate(detection_name = glue::glue("{call} \n ( {total_tissue} )"))



detection_table %>% 
  ggplot(aes(x = call, y = detection_rate,fill = call)) + 
  geom_col(show.legend = F,position = 'dodge2') + 
  facet_wrap(~stmn2_psi_groups) +
  scale_fill_manual(values = c("#88CCEE","#44AA99","#1B7739")) + 
  scale_y_continuous(lim = c(0,0.5),labels = scales::percent) + 
  ylab("Percent of Samples \n UNC13A Cryptic Detected") + 
  theme(text = element_text(size = 18))  +
  ggpubr::theme_pubr() + 
  geom_text(aes(label = total_tissue,y = 0),vjust = 1,size = 6) + 
  theme(text = element_text(size = 18)) +
  ggpubr::theme_pubr() +
  xlab("N Tissues") 
```

UNC13A cryptic PSI 

```{r}
clean_data_table %>%   
  filter(disease_tissue == T) %>% 
  filter(disease_group2 %in% c("FTLD-TDP","ALS-TDP")) %>% 
    mutate(call = fct_relevel(call,
                              "C/C", "C/G", "G/G")) %>%
    filter(stmn2_annotated > 10) %>% 
    filter(cryptic_psi > 0) %>% 
    ggbarplot( x = "call", 
               y = "cryptic_psi", 
               color = 'call',
              position = position_dodge(0.8),
              add = c("mean_se","jitter")) + 
    ylab("UNC13A Cryptic PSI") + 
    scale_color_manual(values = c("#88CCEE","#44AA99","#105e2a")) +
    theme(text = element_text(size = 20)) + 
    stat_compare_means(aes(group = call), label = "p.format") + 
    stat_compare_means(aes(group = call),comparisons = genotype_comparisons,label = "p.signif")
```


```{r}
clean_data_table %>%   
  filter(disease_tissue == T) %>% 
  filter(disease_group2 %in% c("FTLD-TDP","ALS-TDP")) %>% 
    mutate(call = fct_relevel(call,
                              "C/C", "C/G", "G/G")) %>%
    mutate(stmn2_psi_groups = as.numeric(cut_interval(log10(stmn_2_cryptic_psi), n = 2))) %>%
    mutate(stmn2_psi_groups = ifelse(is.na(stmn2_psi_groups),"  No STMN2",stmn2_psi_groups)) %>%
    mutate(stmn2_psi_groups = case_when(stmn2_psi_groups == 1 ~ " Low STMN2",
                                        stmn2_psi_groups == 2 ~ "High STMN2",
                                        TRUE ~ stmn2_psi_groups)) %>%
    filter(cryptic_psi > 0) %>%
    ggbarplot( x = "call", 
               y = "cryptic_psi", 
               color = 'call',
               facet.by = 'stmn2_psi_groups',
              position = position_dodge(0.8),
              add = c("mean_se","jitter")) + 
    ylab("UNC13A Cryptic PSI") + 
    scale_color_manual(values = c("#88CCEE","#44AA99","#105e2a")) +
    theme(text = element_text(size = 20)) + 
    stat_compare_means(aes(group = call), label = "p.format") + 
    stat_compare_means(aes(group = call),comparisons = genotype_comparisons,label = "p.signif")  + 
  xlab("")
```



## Variant allele expression in TDP-43 KD experiments

Across our KD's, we observed a clear allelic bias in the NB cells

```{r}
kd_vafs = fread(file.path(here(),"data","all_kds_unc13.snp.out.tsv"))

kd_vafs %>% 
    left_join(rel_rna_cryptic_amount) %>% 
    filter(!is.na(condition)) %>% 
    mutate(VAF = alt_reads / total) %>% 
    filter(condition != "control" ) %>% 
    filter(source %in% c("sh_dzap","shsy5y_normed","ipsc_normed_ward")) %>% 
    ggplot(aes(x = TARDBP,y = VAF, fill = source,size = total)) + 
    geom_point(pch = 21)  + 
    ylim(0,1)
    
kd_vafs %>% 
    left_join(rel_rna_cryptic_amount) %>% 
    filter(!is.na(condition)) %>% 
    filter(condition != "control" ) %>%     
    mutate(VAF = alt_reads / total) %>% 
    filter(source %in% c("sh_dzap","shsy5y_normed","ipsc_normed_ward")) %>% 
    dplyr::select(sample,ref_reads,alt_reads,TARDBP,source,VAF) %>% 
    data.table::melt(id.vars = c("sample","TARDBP",'source','VAF')) %>% 
    mutate(sample = fct_reorder(sample,-TARDBP)) %>% 
    ggplot(aes(x = sample,y = value, fill = variable)) + 
    geom_col(pch = 21, size = 4) +
    coord_flip()
```

Also look at the coverage in the Liu nuclear facs sorted neurons

```{r}
liu_vafs = fread(file.path(here(),"data","liu_facs_neurons_unc13.snp.out.tsv"))
liu_vafs %>% 
    mutate(VAF = alt_reads / total) %>% 
    data.table::melt(id.vars = c("sample",'VAF','total')) %>% 
    mutate(allele = ifelse(variable == "ref_reads", "C","G")) %>% 
    mutate(sample_name = sub("_TDP_.*", "", sample)) %>% 
    mutate(sample_name = gsub("_unsorted", "", sample_name)) %>% 
    mutate(condition = sub(".*_", "", sample)) %>% 
    ggplot(aes(x = condition,y = value, fill = allele)) + 
    geom_col(pch = 21, size = 4) +
    facet_wrap(~sample_name,scales = "free_y")
```

